docs/src/main/sphinx/release/release-354.md

```diff
@@ -0,0 +1,56 @@
+# Release 354 (XXXXXXXX Mar 2021)
+
+## General
+
+* Preferred write partitioning is now triggered automatically when the number
+  of written partitions exceeds or is equal to
+  ``preferred-write-partitioning-min-number-of-partitions``. The number of written
+  partitions is estimated from stats, therefore CTAS or insert source tables needs to have
+  statistics collected. To forcefully enable preferred partitioning,
+  ``preferred-write-partitioning-min-number-of-partitions`` needs to be set to
+  ``0`` or ``1``. To fully disable preferred write partitioning, set
+  ``use-preferred-write-partitioning`` to ``false``.
+  ({issue}`6920`)
+* Improve performance of LEFT, RIGHT and FULL JOIN when one side of the join is known to produce a single row. ({issue}`7090`)
+  
+## Security
+
+* Fix `Oauth2` repeated authentication, after previous token expires. ({issue}`7172`)
+* Support Oauth2 authentication in Trino CLI. ({issue}`7054`)
+
+## Elasticsearch connector
+
+* Fix failure when reading single-valued fields for array types. ({issue}`7012`)
+
+## Hive connector
+
+* Fix possible S3 connection leaks when using AVRO table format ({issue}`7178`)
+* Use VARCHAR as type for all columns, for tables using CSV SerDe, when Glue metastore is in use ({issue}`7059`)
+* Rename ``hive.parallel-partitioned-bucketed-inserts`` feature config
+  to ``hive.parallel-partitioned-bucketed-writes`` ({issue}`7259`)
+* Fix incorrect query results when reading from an incorrectly bucketed table created by Spark. ({issue}`6848`)
+  
+## Iceberg connector
+
+* Fix queries on empty tables without a snapshot id created using Spark. ({issue}`7233`)
+* Upgrade to Iceberg ``0.11.0``. It is recommended to also upgrade the version in Iceberg readers
+  that may read tables written by Trino. This is to avoid issues with partition transforms for timestamps
+  representing negative epoch values. ({issue}`7049`)
+
+## MemSQL connector
+
+* Add support for MemSQL 3.2. ({issue}`7179`)
+
+## MySQL connector
+
+* Use proper column type (`datetime(3)`) in MySQL when creating a table with `timestamp(3)` column.
+  Previously, the second fraction was being truncated. ({issue}`6909`)
+  
+## PostgreSQL connector
+
+* Fix potentially incorrect query results for queries with `ORDER BY` on a `char`/`varchar` column and `LIMIT`
```

Comment:
nit: `query results for queries` --> `results for queries`

Reply:
There is a additional change and new wording offered in https://github.com/trinodb/trino/issues/7186#issuecomment-800897987

> ```
> * Fix incorrect query results when using `ORDER BY ... LIMIT` clause when sorting on  `char` or `varchar` > columns
>   and `topn-pushdown.enabled` configuration property is enabled. The optimization is now enabled by default.
>   ({issue}`7170`, {issue}`7314`)
> ```

Reply:
"when ... when ..." sounds funny, though. Let me think about a better way to phrase it.

Reply:
I agree... We could have some common vocabulary recommendation doc. cc @electrum

Reply:
I rephrased as "for ORDER BY...LIMIT clause when sorting on ..."